CMSgov/mint-app
# [EASI-2371] - Select a solution

# EASI-2371

## Changes and Description

- Select a solution form
- Changes to props for `AskAQuestion` for solution form
- Mapping of operational needs to route, translations, etc
- Component for rendering opNeed question and answer - `NeedQuestionAndAnswer`
- Component for solution checkbox - `CheckboxCard`
- Began cypress testing for IT Solutions

![Screenshot 2022-11-08 at 10 17 59 AM](https://user-images.githubusercontent.com/95709965/200603258-9654ce94-613b-49b1-a8e9-f6f7fc3aa535.png)

## Notes

- `NeedQuestionAndAnswer` proved to be a task to make dynamic.  Included use of apollo client `directives` to conditionally define query parameters and return type.  This in effect made TS implementation a bit more difficult
- Solution `description` data is not present - using lorem ipsum
- Two operational needs require no answer and will always be in the needs table
- Warning tags to return to tracker only appear after first saving/selecting an op need answer

## How to test this change

- Enter into a model plan and answer an op need questions (ex: opEvalAndLearning - 'Do you plan to use a helpdesk?') 
- Go to IT Solutions tracker from task list and verify the need has moved to the first table
- Click 'Select a solution' for the table row for Help Desk
- Toggle 'Why is this a part of my operational needs?' and verify the question and answer are accurately reflected
- Check one or more checkboxes and select 'Continue'
- Verify the solution/s is now inserted into the table

## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
